### ISSUE TITLE
permanent breakdown of the interface to the server

### ISSUE THREAD

**Issue by Pirol62** [sebr/bhyve-home-assistant#9]
HA 107.7  
bHyve 0.1.10

Once a day, I have these errors:
````
Logger: custom_components.bhyve.pybhyve.websocket
Source: custom_components/bhyve/pybhyve/websocket.py:127
First occurred: 3. April 2020, 18:27:01 (1 occurrences)
Last logged: 3. April 2020, 18:27:01

Websocket exception:
````
````
Logger: custom_components.bhyve.pybhyve.websocket
Source: custom_components/bhyve/pybhyve/websocket.py:131
First occurred: 3. April 2020, 19:19:54 (1 occurrences)
Last logged: 3. April 2020, 19:19:54

Unexpected error Cannot connect to host api.orbitbhyve.com:443 ssl:None [None]
````
Only a HArestart can reestablish the interface

**Comment by sebr** [comment]
This is likely because your OpenSSL implementation doesn't support weak ciphers.

Unfortunately, `api.orbitbhyve.com` rates a big fat red **F**: https://www.ssllabs.com/ssltest/analyze.html?d=api.orbitbhyve.com&hideResults=on

<img width="1103" alt="Screen Shot 2020-04-06 at 2 38 48 pm" src="https://user-images.githubusercontent.com/81972/78523553-7ffa2380-7814-11ea-9c6d-ba4f0fec7ee0.png">

**Comment by Pirol62** [comment]
Hmmm, thanks for the information. As this is not really my knowledge area I do have some difficulties to get my learnings out of that. Do I have to change something or do I have to live with that?
Maybe I have to implement a fallback because I automated my irrigation starting next month. If it is not reliable, then I need a push-message in order to start irrigation via bHyve web interface.

**Comment by sebr** [comment]
What system are you running HA on?

**Comment by Pirol62** [comment]
Rasp 3+ and Hassio

**Comment by sebr** [comment]
No idea mate, sorry. I have no experience with Hassio. I think it's related to your deployment, but I also find this quite odd that it only happens occasionally. Can you please [enable debug logging](https://github.com/sebr/bhyve-home-assistant#debugging) and see if there are any patterns you can ascertain?

**Comment by Pirol62** [comment]
Ok, will do that as soon as I'm in home Office. It could be, that the problem is in my DSL connection. I had some strange interruptions during the last two weeks in connection with fluctuations in bandwidth. I will keep that under investigation.

**Comment by sebr** [comment]
Ah that's interesting. I'll see what I can do about putting in reconnection attempts.

**Comment by Pirol62** [comment]
Hi, I had this error never again but:
I had to replug the bHyve-Hub, so the connection was obviously broken. After the device had reconnected to my wireless lan, the battery sensor returns available, the switch and the raindelay sensor not. I had to restart HA. So your idea to implement some reconnection functionality in case of broken contact might be the right idea :-)
If you like, we can close this issue as the error didn't appear again.
